src/backend/distributed/metadata/metadata_utility.c

```diff
@@ -618,15 +685,70 @@ GenerateShardNameAndSizeQueryForShardList(List *shardIntervalList)
 		char *shardQualifiedName = quote_qualified_identifier(schemaName, shardName);
 		char *quotedShardName = quote_literal_cstr(shardQualifiedName);
 
-		appendStringInfo(selectQuery, "SELECT %s AS shard_name, ", quotedShardName);
-		appendStringInfo(selectQuery, PG_RELATION_SIZE_FUNCTION, quotedShardName);
+		if (includeShardMinMax)
```

Comment:
We don't see to be "including" min max, but rather using a different query.

Maybe "useShardMinMaxQuery"?